### PR TITLE
feat: LNv1 Payment Events

### PR DESCRIPTION
**Pull request by m1sterc001guy**
Similar to https://github.com/fedimint/fedimint/pull/6351, this PR adds payment events for LNv1. It allows us to compute profit, fees, and latency for LNv1.

The LNv2 events are a bit better structured, but this still gives us the necessary data. Since we will be supporting LNv1 for a bit I figured it would be good to add this as well (might even be fun to compare with LNv2).

### CODE REVIEW

File: gateway/ln-gateway/src/state_machine/events.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+use fedimint_ln_common::contracts::ContractId;
+use serde::{Deserialize, Serialize};
+
+use super::pay::OutgoingPaymentError;
+
+#[derive(Serialize, Deserialize)]
+pub struct OutgoingPaymentStarted {
+    pub contract_id: ContractId,
+    pub invoice_amount: Amount,
+}
```
Comment (elsirion) [inline gateway/ln-gateway/src/state_machine/events.rs]: Should contain operation id so e.g. invoice can be looked up from operation log.

**Review by elsirion — COMMENTED** [review]
Looks good, but probably best to include the operation id so additional meta data can be looked up.

File: gateway/ln-gateway/src/state_machine/events.rs (hunk 1 above)
Comment (m1sterc001guy) [reply gateway/ln-gateway/src/state_machine/events.rs]: Fixed!

Comment (elsirion) [reply gateway/ln-gateway/src/state_machine/events.rs]: You only added the operation id to the start event, which is ok I guess as long as consumers keep an index `payment_hash -> operation_id`. Worth considering the storage/ease of consumption trade-off though, cc @dpc.

**Review by elsirion — APPROVED** [review]

**Pull request merged**